### PR TITLE
Refactor post, put and delete requests

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr refactors how we handle post, put and delete requests in our frontend by extracting all api calls to `/util/api`. Toasts are removed as we are currently working on proper error handling that will be implemented in a future pr.

### CODE REVIEW

**Review by TomRomeo — APPROVED** [review]
Looks good, I just found a few small issues.

**Review by memeToasty — APPROVED** [review]

File: frontend/components/flashcard/Flashcard.tsx [hunk 1]
```diff
@@ -10,28 +10,27 @@ import {
...
+import { useDeckDueCards } from "@/util/swr";
 
 interface FlashcardProps {
 	/**
 	 * Unique identifier
 	 */
 	id: string;
 	/**
-	 * Flashcard
+	 * deckID
```
Comment (4KevR) [inline frontend/components/flashcard/Flashcard.tsx:27]:
```suggestion
	 * Deck ID
```

File: frontend/components/group/Member.tsx [hunk 2]
```diff
@@ -84,21 +90,23 @@ export default function Member({
...
-											inviteUser(
-												user.userEmail ?? "",
-												false
-											);
-										}}
+										onClick={() =>
+											user.groupID &&
+											declineGroupRequest(
+												user.groupID,
+												user.userEmail ?? ""
```
Comment (4KevR) [inline frontend/components/group/Member.tsx:107]: What happens when `user.userEmail` contains a falsy value and an empty string is passed to the API? Is this intended?

File: frontend/components/group/Member.tsx [hunk 3]
```diff
@@ -110,9 +118,13 @@ export default function Member({
...
 								<X
 									className="cursor-pointer"
-									onClick={() => {
-										inviteUser(user.userEmail ?? "", false);
-									}}
+									onClick={() =>
+										user.groupID &&
+										declineGroupRequest(
+											user.groupID,
+											user.userEmail ?? ""
```
Comment (4KevR) [inline frontend/components/group/Member.tsx:125]: s.a. for reference

File: frontend/components/navigation/Tabs/UserSettingsTab.tsx [hunk 4]
```diff
@@ -68,7 +65,7 @@ export const UserSettingsTab = ({
 						}}
 						onClick={() => {
 							userNameInputAction.current &&
-								modifyUser(userNameInputAction.current);
+								modifyUser([userNameInputAction.current]);
```
Comment (4KevR) [inline frontend/components/navigation/Tabs/UserSettingsTab.tsx:68]: The function `modifyUser` is currently only used with one parameter. Will there be future use-cases with multiple users to require an array?

File: frontend/util/api.ts [hunk 5]
```diff
@@ -1,23 +1,251 @@
...
+
+// handle group requests
+async function groupRequest(
+	groupID: string,
+	request: (url: string, body?: string) => Promise<Response>
+) {
+	const response = await request(requestGroupRoute(groupID));
+	if (response?.ok)
+		mutateAll([
+			`/api/user/invitations`,
```
Comment (4KevR) [inline frontend/util/api.ts:124]: Why do you not use `invitationsUserRoute`?

**Review by 4KevR — COMMENTED** [review]
Sorry for my late PR review. Changes look great, I just have some comments for clarification.

File: frontend/components/flashcard/Flashcard.tsx (hunk 1 above)
Comment (Julian702) [reply frontend/components/flashcard/Flashcard.tsx]: Fixed in 0aad79af343dae525356927d530582da7f3f0d11

File: frontend/components/group/Member.tsx (hunk 3 above)
Comment (Julian702) [reply frontend/components/group/Member.tsx]: Fixed in 0aad79af343dae525356927d530582da7f3f0d11

File: frontend/util/api.ts (hunk 5 above)
Comment (Julian702) [reply frontend/util/api.ts]: fixed in 0aad79af343dae525356927d530582da7f3f0d11

**Pull request merged**